### DISCUSSION TITLE
How to get reverse related objects using Schema?

### DISCUSSION THREAD

**Discussion by EMUNES** [vitalik/django-ninja#154] [Q&A]
It's pretty convenient to return objects with nested foreign-key objects using ninja. However, I find no tutorial to return objects including reverse related objects using ninja.

Say I have a Django PostComment model whose foreign-key field points to Post object. And for API I want to return a list of Post objects each includes all comments that points to it. Is it possible to do so using Schemas and if so how should I customize the api code?

I'm new to ninja, Thanks for any hint :)

**Comment by vitalik** [comment]
Hi @EMUNES 

Try the folloing:


```Python
from typing import List
from ninja import Schema, Field

class CommentSchema(Schema):
    # define your comment attrs here:
    id: int
    message: str

class PostSchema(Schema):
    id: str
    title: str
    comments: List[CommentSchema] = Field(..., alias='postcomment_set')  # postcomment_set is based on your model or related attr



@api.get('/posts', response=List[PostSchema])
def list_posts(reqiest):
     return Post.objects.all() 
```

**Reply by rngallen** [reply]
Hello I want to use `pre_fetch_related` and `select_related` to reduce number of database queries
My schema are as follows
```
class OrderOutSchema(Schema):
    id: int
    customer: CustomerOrderSchema
    order_number: str
    order_date: date
    reference: str
    product: ProductOrderSchema
    vessel: VesselOrderSchema
    quantity: Decimal
    trucks: int
    created: datetime
    updated: datetime
    created_by: CreatedByOrderSchema
    order_lines: List[OrderLineOutSchema]

class OrderLineOutSchema(Schema):
    id: int
    order_id: int
    line: int
    driver: DriverOrderLineSchema
    truck: TruckOrderLineSchema
    loading_date: date
```
my query is 
```def get_order(request, order_number: str = "", page: int = 1, per_page: int = 10):
    qs = Order.objects.all().select_related("customer", "product", "vessel", "created_by").prefetch_related("order_lines") 
    if order_number:
        qs = qs.filter(Q(order_number__icontains=order_number))
    offset = (page - 1) * per_page
    qs = qs[offset : offset + per_page]
    return HTTPStatus.OK, qs
```
but  how can I use select _related on ` OrderLineOutSchema`  for driver(DriverOrderLineSchema) and truck(TruckOrderLineSchema)??? Other wise for every line on I should query Truck Model and Driver Model respectively.

**Reply by EMUNES** [reply]
Thanks a lot. I didn't expect that I can get reverse objects simply by declaring them in Schema. This is really convenient. I will definitely try this in my future project.

**Comment by jet10000** [comment]
Please add it to document

**Reply by quroom** [reply]
I think so.

**Comment by mohanish2504** [comment]
Does nested objects support support async?

**Comment by MiltosD** [comment]
I tried `comments: List[CommentSchema] = Field(..., alias='postcomment_set')` and got 

`Input should be a valid list [type=list_type, input_value=<django.db.models.fields....bject at 0x7f3d69345110>, input_type=create_reverse_many_to_one_manager.<locals>.RelatedManager]`

**Reply by vitalik** [reply]
@MiltosD need to see more of your code
